### DISCUSSION TITLE
After upgrade to 1.22 - settings page shows: Internal Server Error

### DISCUSSION THREAD

**Discussion by gsigrisi** [teslamate-org/teslamate#1466] [Q&A]
**Describe the bug**

After upgrade to 1.22 - settings page shows: Internal Server Error

**Expected behavior**

**How to reproduce it (as minimally and precisely as possible)**:

**Relevant entries from the logs**

```
teslamate_1  | Request: GET /settings
teslamate_1  | ** (exit) an exception was raised:
teslamate_1  |     ** (UndefinedFunctionError) function nil.id/0 is undefined
teslamate_1  |         nil.id()
teslamate_1  |         (teslamate 1.22.0) lib/teslamate_web/live/settings_live/index.ex:184: anonymous fn/2 in TeslaMateWeb.SettingsLive.Index.prepare/1
teslamate_1  |         (elixir 1.11.3) lib/enum.ex:2193: Enum."-reduce/3-lists^foldl/2-0-"/3
teslamate_1  |         (teslamate 1.22.0) lib/teslamate_web/live/settings_live/index.ex:16: TeslaMateWeb.SettingsLive.Index.mount/3
teslamate_1  |         (phoenix_live_view 0.15.4) lib/phoenix_live_view/utils.ex:320: anonymous fn/4 in Phoenix.LiveView.Utils.maybe_call_live_view_mount!/4
teslamate_1  |         (telemetry 0.4.2) /opt/app/deps/telemetry/src/telemetry.erl:262: :telemetry.span/3
teslamate_1  |         (phoenix_live_view 0.15.4) lib/phoenix_live_view/static.ex:306: Phoenix.LiveView.Static.call_mount_and_handle_params!/5
teslamate_1  |         (phoenix_live_view 0.15.4) lib/phoenix_live_view/static.ex:131: Phoenix.LiveView.Static.render/3```



**Environment**

- TeslaMate version: 1.22
- Type of installation: Docker
- OS TeslaMate is installed on:
- User OS & Browser:
- Others:

**Comment by adriankumpf** [comment]
Have you by any chance changed or deleted data in the database yourself? I can't see any connection with the changes made in v1.22 right now

**Comment by gsigrisi** [comment]
Hi, thanks for the feedback. 
I just deleted from the DB a state which was reporting errors. I followed the procedure described in one of the git issues comment to correctly delete it. 

But this happened 2 updates ago. 
Thanks for your feedback and amazing work!!

**Comment by adriankumpf** [comment]
I'm not sure what you mean by a state that was reporting errors. From the error message it looks like a car was deleted from the `cars` table but the corresponding entry in the `car_settings` is still present. 

I'll move this to the Q&A section for now because I don't see any signs of a bug so far.

**Comment by gsigrisi** [comment]
Hi, this is the link I followed to safely remove the setting from the postgres db.

https://github.com/adriankumpf/teslamate/issues/706

**Comment by drewpost** [comment]
I am experiencing this issue too on 1.23 and now on 1.23.1. It persists after a restart and update. I have NOT modified the database in anyway and am running via Docker install on a Raspberry Pi. cc @adriankumpf  Did your investigations get anywhere? Happy to provide logs etc if you tell me where to get them from

**Reply by adriankumpf** [reply]
Well, as I said, the reason is (was) a manual modification of the database. You can check if the number of row in the "cars" table matches those in "car_settings". 

If you didn't make any modifications, could this be a sign for a bad SD card?

**Reply by gsigrisi** [reply]
Hi Adrian, Drew,
I rechecked the DB and I found the issue @adriankumpf was referring to. 

```
teslamate=# SELECT id FROM car_settings;
 id
----
  2
  1
  3
(3 rows)

teslamate=# SELECT id FROM cars;
 id
-----
   1
 161
(2 rows)
```

By the look at this output, it seems I have an inconsistency. 

```
teslamate=# SELECT id, model FROM cars;
 id  | model 
-----+------
   1 | 3     
 161 | S    
(2 rows)
```

The car with ID=161 was a loaner, the car with id=1 is my current Tesla. 

The ID=2 and ID=3 should not be car_settings table instead ID=161 should be in and it is not there. 

I followed the instructions from here to remove the car I didn't want:
https://github.com/adriankumpf/teslamate/issues/706

Thanks for your comments.
